### ISSUE TITLE
Create a new wiki page entry

### ISSUE THREAD

**Issue by media1mogul** [media1mogul/CSCIE33a#9]
# Story Description

Let the user create a new page from a link in the sidebar.

_[[GIVE ADDITIONAL CONTEXT/LINKS HERE]]_

# Definition of Done
- [x] Clicking “Create New Page” in the sidebar should take the user to a page where they can create a new encyclopedia entry.
- [x] Users should be able to enter a title for the page and, in a textarea, should be able to enter the Markdown content for the page.
- [x] Users should be able to click a button to save their new page.
- [x] When the page is saved, if an encyclopedia entry already exists with the provided title, the user should be presented with an error message.
- [x] Otherwise, the encyclopedia entry should be saved to disk, and the user should be taken to the new entry’s page.
  
# Review/QA Test Process

- **Link to PR**: _Whoever works this issue fills in this part_

_Whoever works this issue fills in this part_

# Does this PR introduce a user interface change?
 
_Whoever works this issue fills in this part_

_If mockups are available, please link/include them here_

**Comment by media1mogul** [comment]
I'm immediately running into the same issue of knowing what I have and what I don't have for pages.
- The first thing that comes to mind is again the use of a dictionary to know what pages I have 

The trouble with this is that now I need  CRUD operations for the dictionary.
And when would it be initialized?

The rest of the operations would be easy. Just mimick what happens to the wiki page life cycle.

There must be an init portion to this code somewhere.

-----
Ok so I can [initialize it in apps.py](https://pythonin1minute.com/where-to-put-django-startup-code/)
----
OK - but do I need it?

**Comment by media1mogul** [comment]
OK quick look at wikipedia as an example.  They do not force all caps for page names.
Also - if this were a wiki - there will be others working on the same wiki.  What we're doing is a far cry from a real multi-use site, but still what if while I was doing one thing, someone else already added the page I wanted to add. Passing around a dictionary is not going to work with that. The data in a dictionary of page names will get old too quick. Better to re-query each time i need the list. - At least at this small scale

**Comment by media1mogul** [comment]
- [ ]  Clicking “Create New Page” in the sidebar should take the user to a page where they can create a new encyclopedia entry.
  - Create a very simple form to start that jsut lets you put in the title.
  -

**Comment by media1mogul** [comment]
Need to handle:
- Failed form validatino
- page exists.

**Comment by media1mogul** [comment]
I played around with re-displaying the content with a message if the file already exists. I was hoping this would be easy. This way th user won't lose their data.  It may be easy but I'm not ready for it yet.

**Comment by media1mogul** [comment]
It seems clear now that I need an error.html page.
It will solve this as well as the failed validation problem as well as there is not wiki page here.

**Comment by media1mogul** [comment]
I've got an error handler, and along the way I learned more about how the framework works.

I don't think my error handler is ideal. I used URL based parameters to communicate between the view where I wanted to originate the error and the view where I wanted to display the error.

I was thinking at first that I could create an error html file on disk. Maybe even in markdown form. However I rejected this because I'd have to have CRUD operations for that page -  but maybe not?  Maybe I'd always use error.html and just overwrite the last instance of the file each time.

This would let me preserve the person's editing in the case of an error.

Also the way that I do it looks yucky.
http://localhost:8000/wiki/error?title=Error&bold_text=There%20is%20no%20Wiki%20entry%20for:%20biff&body=.
